student/src/main/java/tds/student/configuration/RedisCacheConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package tds.student.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.util.StringUtils;
+import tds.dll.common.performance.caching.impl.JedisSentinelConnectionFactory;
+import tds.dll.common.performance.caching.impl.NameAwareRedisCacheManager;
+import tds.dll.common.performance.caching.impl.RedisJsonSerializer;
+
+import java.util.Set;
+
+/**
+ * This configuration initializes a Redis Sentinel-backed cache manager if the "redisCaching" profile is active.
+ */
+@Profile("redisCaching")
+@Configuration
+public class RedisCacheConfiguration {
```

Comment:
This will enable redis caching iff the server is run with a "redisCaching" profile.